jina/peapods/runtimes/grpc/__init__.py

```diff
@@ -217,6 +216,16 @@ def _handle(self, msg: Message) -> Message:
 
         return msg
 
+    def _get_expected_parts(self, msg):
+        if msg.is_data_request:
+            if not self._static_routing_table:
+                graph = RoutingTable(msg.envelope.routing_table)
+                return graph.active_target_pod.expected_parts
+            else:
```

Comment:
Are we sure if static routing table, that `num_part` will always be properly populated?

Reply:
I think so. At least for all non shared Pods (non external/regular) we know the information beforehand. For head peas it should be derived from `needs` and for tail peas it should be derived from `parallel`. As far as I can see it should be correct. But it might be worth adding more tests checking this, that is true

Reply:
Actually it did not work for the `parallell` case because the pea args were not updated after the Flow had set the Pod args. I fixed this and added tests for this case.